db/view/view.cc

```diff
@@ -2645,6 +2646,30 @@ update_backlog node_update_backlog::fetch() {
     return std::max(fetch_shard(this_shard_id()), _max.load(std::memory_order_relaxed));
 }
 
+future<std::optional<update_backlog>> node_update_backlog::fetch_if_changed(std::optional<db::view::update_backlog> last_backlog) {
+    auto np = co_await map_reduce(boost::irange(0u, smp::count),
+            [this] (shard_id shard) {
+                return smp::submit_to(shard, [this, shard] {
+                    if (_backlogs[shard].need_publishing) {
+                        _backlogs[shard].need_publishing = need_publishing::no;
+                        return need_publishing::yes;
+                    }
+                    return need_publishing::no;
```

Comment:
Nit: you can use `std::exchange` to make the body of the lambda nice and short:

```c++
return std::exchange(_backlogs[shard].need_publishing, need_publishing::no);
```

Reply:
Ok